diff --git a/src/anbox/graphics/program_family.h b/src/anbox/graphics/program_family.h
index 4de720e..de8ea42 100644
--- a/src/anbox/graphics/program_family.h
+++ b/src/anbox/graphics/program_family.h
@@ -19,6 +19,9 @@
 #ifndef ANBOX_GRAPHICS_PROGRAM_FAMILY_H_
 #define ANBOX_GRAPHICS_PROGRAM_FAMILY_H_

+#include <string>
+#include <sstream>
+#include <iostream>
 #include <map>
 #include <unordered_map>
 #include <utility>