diff --git a/src/anbox/platform/sdl/window.cpp b/src/anbox/platform/sdl/window.cpp
index 752b13c..cd728d1 100644
--- a/src/anbox/platform/sdl/window.cpp
+++ b/src/anbox/platform/sdl/window.cpp
@@ -81,13 +81,13 @@ Window::Window(const std::shared_ptr<Renderer> &renderer,
   SDL_GetWindowWMInfo(window_, &info);
   switch (info.subsystem) {
     case SDL_SYSWM_X11:
-      native_display_ = static_cast<EGLNativeDisplayType>(info.info.x11.display);
-      native_window_ = static_cast<EGLNativeWindowType>(info.info.x11.window);
+      // native_display_ = static_cast<EGLNativeDisplayType>(info.info.x11.display);
+      // native_window_ = static_cast<EGLNativeWindowType>(info.info.x11.window);
       break;
 #if defined(WAYLAND_SUPPORT)
     case SDL_SYSWM_WAYLAND:
-      native_display_ = static_cast<EGLNativeDisplayType>(info.info.wl.display);
-      native_window_ = reinterpret_cast<EGLNativeWindowType>(info.info.wl.surface);
+      // native_display_ = static_cast<EGLNativeDisplayType>(info.info.wl.display);
+      // native_window_ = reinterpret_cast<EGLNativeWindowType>(info.info.wl.surface);
       break;
 #endif
 #if defined(MIR_SUPPORT)